src/BugsnagUnity/Native/Android/NativeInterface.cs

```diff
@@ -658,19 +679,25 @@ public void AddMetadata(string section, IDictionary<string,object> metadata)
             {
                 return;
             }
+            var disposableContainer = new DisposableContainer();
             CallNativeVoidMethod("addMetadata", "(Ljava/lang/String;Ljava/util/Map;)V",
-            new object[] { section, DictionaryToJavaMap(metadata) });           
+            new object[] { section, DictionaryToJavaMap(metadata, ref disposableContainer) });
+            disposableContainer.Dispose();
         }
 
-        internal static AndroidJavaObject DictionaryToJavaMap(IDictionary<string, object> inputDict)
+
+
+        internal static AndroidJavaObject DictionaryToJavaMap(IDictionary<string, object> inputDict, ref DisposableContainer disposable)
```

Comment:
This doesn't need to be a pass by reference.  It would only need to be so if this method was intending to replace the reference and have that reflected to the caller.

Reply:
Also I think `disposables` would be a better parameter name.